### PR TITLE
Fix timeline breakage caused by change of flutter event name.

### PR DESCRIPTION
**Pull request by kenzieschmoll**
The event that parented the UI portion of a Flutter frame used to be 'VSYNC'. Once I upgraded my Flutter checkout, the event was named 'VsyncProcessCallback'. In the few places of the Timeline that we depend on this, we now check for the old name 'VSYNC' and the new name 'VsyncProcessCallback'.

There's an open issue to try to get tests in the engine that keep these event names stable: https://github.com/flutter/flutter/issues/27609

### CODE REVIEW

File: packages/devtools_app/lib/src/timeline/flutter/timeline_controller.dart [hunk 1]
```diff
@@ -275,14 +276,23 @@ class TimelineController implements DisposableController {
     _loadOfflineDataController.add(offlineTimelineData);
   }
 
-  int _threadIdForEvent(
-    String targetEventName,
+  int _threadIdForEvents(
+    List<String> targetEventNames,
```
Comment (devoncarew) [inline packages/devtools_app/lib/src/timeline/flutter/timeline_controller.dart]: Perhaps define this to be a `Set<String>`

File: packages/devtools_app/lib/src/timeline/flutter/timeline_controller.dart [hunk 2]
```diff
@@ -275,14 +276,23 @@ class TimelineController implements DisposableController {
...
     List<TraceEventWrapper> traceEvents,
   ) {
     const invalidThreadId = -1;
     return traceEvents
-            .firstWhere((trace) => trace.event.name == targetEventName,
-                orElse: () => null)
+            .firstWhere(
+              (trace) {
+                for (final target in targetEventNames) {
```
Comment (devoncarew) [inline packages/devtools_app/lib/src/timeline/flutter/timeline_controller.dart]: And here, return `targetEventNames.contains(target.event.name)` (which will be efficient, given the change to a Set).

File: packages/devtools_app/lib/src/timeline/flutter/timeline_processor.dart [hunk 3]
```diff
@@ -40,7 +40,11 @@ StringBuffer debugFrameTracking = StringBuffer();
 
 const String gpuEventName = 'GPURasterizer::Draw';
 
-const String uiEventName = 'VSYNC';
+// For older versions of Flutter, the starting event for the UI portion of a
+// Flutter frame was 'VSYNC'. We need to check for both so that we can still
+// support users on older versions of Flutter.
+const String uiEventNameOld = 'VSYNC';
+const String uiEventName = 'VsyncProcessCallback';
```
Comment (devoncarew) [inline packages/devtools_app/lib/src/timeline/flutter/timeline_processor.dart]: Hmm, `VSYNC` seems more semantic. It'd be good to get this under test in the engine (as your bug mentions).

File: packages/devtools_app/lib/src/timeline/html_timeline_controller.dart [hunk 4]
```diff
@@ -184,14 +185,23 @@ class TimelineController implements DisposableController {
...
+    List<String> targetEventNames,
     List<TraceEventWrapper> traceEvents,
   ) {
     const invalidThreadId = -1;
     return traceEvents
-            .firstWhere((trace) => trace.event.name == targetEventName,
-                orElse: () => null)
+            .firstWhere(
+              (trace) {
+                for (final target in targetEventNames) {
```
Comment (devoncarew) [inline packages/devtools_app/lib/src/timeline/html_timeline_controller.dart]: Same suggestions as above

File: packages/devtools_app/lib/src/timeline/flutter/timeline_controller.dart [hunk 5]
```diff
@@ -252,8 +252,9 @@ class TimelineController implements DisposableController {
 
     // TODO(kenz): once each trace event has a ui/gpu distinction bit added to
     // the trace, we will not need to infer thread ids. This is not robust.
-    final uiThreadId = _threadIdForEvent(uiEventName, traceEvents);
-    final gpuThreadId = _threadIdForEvent(gpuEventName, traceEvents);
+    final uiThreadId =
+        _threadIdForEvents([uiEventName, uiEventNameOld], traceEvents);
```
Comment (devoncarew) [inline packages/devtools_app/lib/src/timeline/flutter/timeline_controller.dart]: `[uiEventName, uiEventNameOld]` ==> `const {uiEventName, uiEventNameOld}` ?

Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/timeline/flutter/timeline_controller.dart]: done

File: packages/devtools_app/lib/src/timeline/flutter/timeline_controller.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/timeline/flutter/timeline_controller.dart]: done

File: packages/devtools_app/lib/src/timeline/flutter/timeline_controller.dart (hunk 2 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/timeline/flutter/timeline_controller.dart]: done

File: packages/devtools_app/lib/src/timeline/html_timeline_controller.dart (hunk 4 above)
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/timeline/html_timeline_controller.dart]: done

**Pull request merged**